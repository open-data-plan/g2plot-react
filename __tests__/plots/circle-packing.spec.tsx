import { render } from '@testing-library/react';
import axios from 'axios';
import React from 'react';
import CirclePackingChart from '../../src/plots/circle-packing';

describe('CirclePackingChart', () => {
  test('should render without crashed', async () => {
    const res = await axios.get('https://gw.alipayobjects.com/os/antfincdn/%24m0nDoQYqH/basic-packing.json');
    const result = render(
      <CirclePackingChart
        {...{
          autoFit: true,
          data: res.data,
          label: false,
          legend: false,
          hierarchyConfig: {
            sort: (a, b) => b.depth - a.depth,
          },
        }}
      />,
    );

    expect(result.container).toMatchSnapshot();
  });
});
