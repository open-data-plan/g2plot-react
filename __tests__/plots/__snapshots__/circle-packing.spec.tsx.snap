// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CirclePackingChart should render without crashed 1`] = `
<div>
  <div
    data-chart-source-type="G2Plot"
    size-sensor-id="1"
  >
    <div
      style="position:relative;"
    >
      <canvas
        height="400"
        style="width: 400px; height: 400px; display: inline-block; vertical-align: middle;"
        width="400"
      />
    </div>
    <object
      class="size-sensor-object"
      data="about:blank"
      style="display: block; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1; opacity: 0;"
      tabindex="-1"
      type="text/html"
    />
  </div>
</div>
`;
